File: Leetcode/268-Missing-Number/MissingNumber.py
class Solution(object):
    def missingNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if nums == None or len(nums) == 0:
            return -1
        res=0
        for i in range(len(nums)) :
            res = res ^ nums[i] ^ (i+1)
        return res
    

File: Leetcode/268-Missing-Number/MissingNumber.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        if (nums.size() == 0) {
            return -1;
        }
        int res = 0;
        for (int i=0; i<nums.size(); i++) {
            res = res ^ nums[i] ^ (i+1);
        }
        return res;
    }
};
